flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-mysql/src/main/java/org/apache/flink/cdc/connectors/mysql/source/reader/MySqlPipelineRecordEmitter.java

```diff
@@ -89,7 +90,16 @@ public MySqlPipelineRecordEmitter(
     protected void processElement(
             SourceRecord element, SourceOutput<Event> output, MySqlSplitState splitState)
             throws Exception {
-        if (isLowWatermarkEvent(element) && splitState.isSnapshotSplitState()) {
+        if (StartupOptions.snapshot().equals(sourceConfig.getStartupOptions())) {
+            // In snapshot mode, we simply emit all schemas at once.
+            createTableEventCache.forEach(
+                    createTableEvent -> {
+                        output.collect(createTableEvent);
+                        TableId tableId = TableId.parse(createTableEvent.tableId().identifier());
+                        alreadySendCreateTableTables.add(tableId);
+                    });
+            alreadySendCreateTableForBinlogSplit = true;
+        } else if (isLowWatermarkEvent(element) && splitState.isSnapshotSplitState()) {
```

Comment:
You may need to rebase to solve the conflict as there is another change in MySqlPipelineRecordEmitter.java that introduced in https://github.com/apache/flink-cdc/commit/eee0cc06f7bf842034d0e22bc45f8ffb811f7f78#diff-7da556d0b9e83dcd71a4a2319c7588c813a5c14ac0dfc180fd096ae7b2a4a5b0.

Reply:
Hi. I'm rebasing and testing.